### DISCUSSION TITLE
نصب کامل warp و حل مشکل کنسول و باگ خوردن هسته

### DISCUSSION THREAD

**Discussion by NimaTarlani** [Gozargah/Marzban#367] [Show and tell]
سلام دوستان. 
من موقه نصب وارپ به باگ هایی میخوردم که کنسول رگباری اونو برمیگردوند.

**امیدوارم این اموزش کمکتون کنه**

قبل از همه وارپ رو نصب میکنین:
`bash <(curl -fsSL git.io/warp.sh) proxy`

بعد تو تسمت outbounds این کد رو وارد میکنید:
`{
      "protocol": "socks",
      "settings": {
        "servers": [
          {
            "address": "127.0.0.1",
            "port": 40000
          }
        ]
      },
      "tag": "WARP"
    }`

برای مثال:

"outbounds": [
    {
      "tag": "freedom",
      "protocol": "freedom",
      "settings": {}
    },
    {
      "protocol": "blackhole",
      "tag": "blackhole"
    },
    {
      "tag": "DNS-Internal",
      "protocol": "dns",
      "settings": {
        "address": "127.0.0.53",
        "port": 53
      }
    },
    {
      "protocol": "socks",
      "settings": {
        "servers": [
          {
            "address": "127.0.0.1",
            "port": 40000
          }
        ]
      },
      "tag": "WARP"
    }
  ]

و بعدش این کد رو تو بخش rules اضافه میکنین:

`{
        "domain": [
          "ipinfo.io",
          "iplocation.net",
          "google.com",
          "spotify.com",
          "whatismyipaddress.com",
          "speedtest.net"
        ],
        "outboundTag": "WARP",
        "type": "field"
      }`

برای مثال:
"rules": [
      {
        "type": "field",
        "outboundTag": "blackhole",
        "ip": [
          "geoip:private",
          "geoip:ir"
        ]
      },
      {
        "type": "field",
        "port": 53,
        "network": "tcp,udp",
        "outboundTag": "DNS-Internal"
      },
      {
        "type": "field",
        "outboundTag": "blackhole",
        "protocol": [
          "bittorrent"
        ]
      },
      {
        "domain": [
          "ipinfo.io",
          "iplocation.net",
          "google.com",
          "spotify.com",
          "whatismyipaddress.com",
          "speedtest.net"
        ],
        "outboundTag": "WARP",
        "type": "field"
      }
    ]
  }

و تمام.

**Comment by M03ED** [comment]
سلام 
این روشی که برای اعمال وارپ استفاده کردید بدترین روش وارپ هست و بسیار مصرف منابع بالایی داره 
روش هایی برای استفاده بهینه از وارپ وجود داره و قبلا تو همین گیت هاب اموزش داده شده میتونید از اون ها استفاده کنید 
اموزش شون هم به صورت نصفه و نیمه الان روی داکیومنت مرزبان قرار داره و نسخه کامل هم الان به داکیومت pr شده و فقط منتظر تایید هست ، میتونید از داکیومنت مرزبان روش ها رو مطالعه کنید
